### PR TITLE
Add image files within content via .resources

### LINKED ISSUE #23: Add local first image files via .Resources
**Is this a BUG REPORT or FEATURE REQUEST?**:
feature

**What happened**:
All static files including fragment images are stored in `/static/<fragment>/` and therefore are not bundled close to the actual content.

**What you expected to happen**:
Enabling bundling of images with actual content via .Ressources

**Anything else we need to know?**:
Needs migration to page bundles first #22.
Local first with a fallback to the /static/ folder.

### PR DESCRIPTION
**Pull request by lauragift21**
**What this PR does / why we need it**: 
Add images within content files while still having static files intact.

**Which issue this PR fixes** *(optional - uncomment and add issue)*:
fixes #23

### CODE REVIEW

**Comment by lauragift21** [comment]
@stp-ip I eventually got the hang of the whole idea but still had difficulties getting it to work also need to verify if I'll have to go through the process for all image resources in a fragment.

**Comment by stp-ip** [comment]
Also you added content/index/hero/images/*, which we don't use and wouldn't be using.
The locations as far as I think would be:
* content/index/hero/* <-- fragment specific (most specific)
* content/index/* <-- page specific (less specific)
* static/* <-- global (least specific)

File: layouts/partials/fragments/hero.html [hunk 1]
```diff
@@ -1,4 +1,8 @@
-{{ $hero := .self.Params  }}
+{{ $hero := .self.Params }}
+{{ $name := .self.Name }}
+{{ $name_t := strings.TrimSuffix "/index.md" $name }}
+{{ $name_t := strings.TrimSuffix ".md" $name_t}}
```
[the remark this answers is not in the document]
Comment (lauragift21) [reply layouts/partials/fragments/hero.html]: Alright. sounds great. If it was possible it'd be better add them into something like a shortcode but figured that only works with markdown content.

File: exampleSite/content/index/hero/index.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+logo = "logo.svg"
+logo_width = "500px" # optional - will default to image width
+#logo_height = "150px" # optional - will default to image height
+
+[[resources]]
+src = "**header.jpg"
+title = "header"
+[[resources]]
+src = "**logo.svg"
+title = "logo"
```
[the remark this answers is not in the document]
Comment (lauragift21) [reply exampleSite/content/index/hero/index.md]: Also missed the page specific integration added that too.

File: exampleSite/content/index/embed-video.md [hunk 3]
```diff
@@ -1,17 +0,0 @@
-+++
```
Comment (stp-ip) [inline exampleSite/content/index/embed-video.md]: Why remove the file?

File: exampleSite/content/index/embed-subscribe.md [hunk 4]
```diff
@@ -1,21 +0,0 @@
-+++
```
Comment (stp-ip) [inline exampleSite/content/index/embed-subscribe.md]: Why remove the file?

File: layouts/partials/fragments/hero.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (stp-ip) [reply layouts/partials/fragments/hero.html]: I was looking into shortcodes and they might actually work for this. I'll try to dive deeper on how to achieve that, but we should probably do that in another PR anyway.

File: exampleSite/content/index/embed-video.md (hunk 3 above)
Comment (lauragift21) [reply exampleSite/content/index/embed-video.md]: This is most likely a mistake from me I wasn't supposed to add it. I changed the name of the file to embed_video for consistency.

**Comment by lauragift21** [comment]
Yeah, I added that too i.e per page resources and all images for hero fragments is also included.  My next commit should make that clear.

File: layouts/partials/fragments/hero.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (lauragift21) [reply layouts/partials/fragments/hero.html]: Great looking forward to it.

**Comment by stp-ip** [comment]
Maybe I'm missing it, but I see the header.jpg being added, but it's not used within the layout file. Only the logo is used and only with static/* or per fragment. Or maybe a commit is missing?

**Comment by lauragift21** [comment]
I also missed that, I've added the specified changes also noticed the header image is not used within the fragment but in the CSS file.

**Comment by stp-ip** [comment]
This needs a follow-on PR to move the header image from CSS to the layout code.

**Comment by stp-ip** [comment]
/LGTM
Additional rough review appreciated from @mpourismaiel

**Comment by mpourismaiel** [comment]
Will review it tonight.

**Comment by mpourismaiel** [comment]
Looks good to me too. However we should use a shortcode in the future to access the images.

**Comment by stp-ip** [comment]
OK let's merge with 2 things:
* a follow-up PR to move the header image from CSS to layout files
* a general issue to move the image code into a shortcode

**Pull request merged**